### DISCUSSION TITLE
Github Secrets not be found

### DISCUSSION THREAD

**Discussion by markgoho** [community/community#25701] [Actions]
<p>Recently (today) it looks like Github Secrets aren’t being passed in properly:</p>
<p>Good run: <a href="https://github.com/dchbx/enroll-ui/runs/545511347?check_suite_focus=true" rel="nofollow noopener">https://github.com/dchbx/enroll-ui/runs/545511347?check_suite_focus=true</a></p>
<p>env:</p>
<p>STATUS: passing</p>
<p>SHA: 708bdd78670da9d35e260524a1c576f4fccc0596</p>
<p>BRANCH: refs/heads/master</p>
<p>YELLR_KEY: ***</p>
<p>YELLR_URL: ***</p>
<p>NX_CLOUD_AUTH_TOKEN: ***</p>
<p>Bad run: <a href="https://github.com/dchbx/enroll-ui/runs/549703106?check_suite_focus=true" rel="nofollow noopener">https://github.com/dchbx/enroll-ui/runs/549703106?check_suite_focus=true</a></p>
<p>env:</p>
<p>STATUS: passing</p>
<p>SHA: 55f69459498064d0af45b48b79def6f28863a75c</p>
<p>BRANCH: refs/heads/master</p>
<p>YELLR_KEY:</p>
<p>YELLR_URL:</p>
<p>NX_CLOUD_AUTH_TOKEN:</p>

**Comment by SAGV** [comment]
<p>I am having the same issue, unfortunately. Some secret values are now blank as of today…</p>

**Comment by kaikue** [comment]
<p>I’m also having this issue. I notice that in your workflow yml file, like in mine, the missing secret is defined in the env section. I have other secrets referenced outside of env (in the jobs section), and those are passed in properly. Weirdly, when I change one of the references to the env variable to directly reference the problematic secret (but keep the other ones still referencing the env variable), everything works. So it seems like Github Actions is looking at the jobs section, checking which secrets are being used, and only passing those in, forgetting to check if they are being used by the env section.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/markgoho">@markgoho</a> <a class="mention" href="/kaikue">@kaikue</a> <a class="mention" href="/sagv">@sagv</a> ,</p>
<p>I also tested using secrets in my workflows, I tried setting secrets as <strong>env</strong> , and also tried directly using secrets in <strong>with</strong> section of an action, however everything works fine on my side. I did not reproduce the problem.</p>
<p>I have helped you report this ticket to the appropriate engineering team for further investigation and evaluation. If they have any progress, I will notify you in time, and sometimes the appropriate engineers may directly reply you here.</p>
<p>In addition, please make sure that the secrets you want to use are still exising in the  <strong>Secrets</strong> list on the <strong>Settings</strong> tab of your repositories. Make sure nobody have removed or changed them.</p>
<p>If the problem still exists, you can try to remove the secrets then re-add them to the list to see if the problem can be solved.</p>

**Comment by markgoho** [comment]
<p>I removed the secrets and re-added them with no success. There’s been no change to our repo, secrets, workflow file, or anything else. It just stopped working between Monday and Tuesday.</p>

**Comment by markgoho** [comment, marked as answer]
<p>BINGO!</p>
<p>I moved the secrets references into the job and they were pulled in perfectly. So, Github Actions has a regression bug. <a class="mention" href="/brightran">@brightran</a> can you flag this please?</p>

**Comment by ericsciple** [comment]
<p>Sorry for the disruption. The behavior will be reverted shortly.</p>
<p>The scenario is the root env in the yaml file references a secret.</p>
<p>Anywhere else in the yaml file that references a secret should work. For example, step-level env. Otherwise let me know if you have another scenario where it was broken.</p>

**Comment by ericsciple** [comment]
<p>Should be reverted now. Can you trigger another run and confirm?</p>

**Comment by kaikue** [comment]
<p>It works for me now. Thanks for the fix!</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/kaikue">@kaikue</a> glad that the problem has gone on your side.</p>
<p><a class="mention" href="/markgoho">@markgoho</a> <a class="mention" href="/sagv">@sagv</a> could you please check if it also back to work on your side?</p>
<p>Apologies for the inconvenience caused to you.</p>

**Comment by markgoho** [comment]
<p>Yes, working now. Thanks for investigating and helping to fix this regression.</p>

**Comment by NAJ8ry** [comment]
<p>Secrets don’t appear to work within the windows-latest or windows-2019</p>
<p>I have two workflows on the same repository. One uses ubuntu-latest and the other windows-latest. They should have the same visibility. When I replace the username and password with literal text it works. Secrets are not functional within the windows environment. Please can you help?</p>
<ul>
<li>name: Docker login<br>
env:<br>
DockerUserName: ${{ secrets.DockerHubUser }}<br>
DockerPassword: ${{ secrets.DockerHubPassword }}<br>
run: |<br>
echo ‘Docker login’<br>
set h+</li>
</ul>
<p>echo “$DockerPassword” | docker login -u “$DockerUserName” --password-stdin <a href="http://index.docker.io" rel="nofollow noopener">index.docker.io</a></p>

**Comment by ericsciple** [comment]
<p><span class="mention">@naj8ryon</span> Windows the “run” step uses powershell by default. Try “$env:DockerPassword” or you can <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell" rel="nofollow noopener">change the shell</a></p>

**Comment by NAJ8ry** [comment]
<p>Brilliant.</p>
<p>The"$env:DockerPassword" approach for getting the variables has worked.</p>
<p>Thank you <a class="mention" href="/ericsciple">@ericsciple</a> . I really appreciate your help.</p>

**Comment by benranck** [comment]
<p><a class="mention" href="/ericsciple">@ericsciple</a> I’ve just run into this issue. When I had the secret defined at the organisation level it wasn’t being picked up by the action (it was initially set to private repo, then I tested by allowing only the specific repo). When I move the secrets to the repo itself the workflow worked correctly.</p>

**Comment by Roms1383** [comment]
<p>I don’t know what’s happening either, I tried defining <code>env</code> vars from secrets in my workflow, they’re empty no matter if I defined them at <code>job</code> or <code>steps</code> levels.</p>
<p>I’ve noted that these are only available on paid accounts for private repo using organization secrets, but I just simply use a private github repo under my own name, with secrets on the repo itself.</p>
<p>Anybody can give me a hint ?</p>

**Comment by Roms1383** [comment]
<p>Ok my apologies here, mistake was mine. I forgot to pipe my env vars to Docker ^^’</p>

**Comment by knsakib** [comment]
<p>This is happening again, today!</p>

**Comment by roryabraham** [comment]
<p>I’m experiencing some flaky jobs due to this too. The same job has been intermittently failing for a couple months (with no changes to the workflow), each time due to the secrets not being available for some unknown reason. I hadn’t noticed because the <code>SLACK_WEBHOOK</code> secret is not available so it’s been failing silently. But today we had three failures.</p>
<p><a href="https://github.com/Expensify/Expensify.cash/actions/workflows/deployBlocker.yml?query=is%3Asuccess" rel="noopener nofollow ugc">All the successful workflow runs</a></p>
<p><a href="https://github.com/Expensify/Expensify.cash/actions/workflows/deployBlocker.yml?query=is%3Afailure" rel="noopener nofollow ugc">All the intermittently failing workflow runs</a></p>
<p>IDK if this is useful debugging information, but it seems like a failed job will consistently fail again, even if retried. The job runs on <code>ubuntu-latest</code>, could this be related to the <code>20.04</code> migration?</p>

**Comment by feross** [comment]
<p>I’ve also noticed that secrets are missing randomly on certain test runs:</p>
<p>When it’s working, I see:</p>
<pre><code class="lang-auto">  env:
    B2_APPLICATION_KEY: ***
    B2_APPLICATION_KEY_ID: ***
</code></pre>
<p>When it randomly fails, I see:</p>
<pre><code class="lang-auto">  env:
    B2_APPLICATION_KEY:
    B2_APPLICATION_KEY_ID:
</code></pre>
<p>Note the missing secrets indicated by no <code>***</code> in the output.</p>
<p><a class="mention" href="/brightran">@brightran</a> Can you escalate this to the appropriate engineering team?</p>

**Reply by frezbo** [reply]
This seems to be still an issue, seems like it takes some time for the secrets to porpogate, one of the repos it works, not in another. Exactly same action yaml

**Reply by mattvb91** [reply]
currently having this exact issue of missing secrets even tho they have been defined and working for months. All of a sudden their missing

**Reply by ghost** [reply]
Same here, I am going nuts

**Reply by mattvb91** [reply]
whats weird is it seems to be new branches that have lost access to old secrets. My develop branch still works fine but new branches cant access the same secret even with no changes.

**Reply by ghost** [reply]
I am using Environment secrets, which work perfectly fine with same exact job definition _in a different repository_.

**Reply by makslevental** [reply]
@NSimpragaVolur do you have a fix? Did it start working eventually? I have the same thing - same job definition, different repo, same secret names, one repo works, the other has empty secrets.

**Reply by ghost** [reply]
I just edited my pipelines so they dynamically fetch the tokens from the specified Static Webapp, and used OIDC to authenticate the actions principal, a managed identity. OIDC is secretless itself so I circumvented the need to define any static secrets.  

This is a good guide: https://www.cloudwithchris.com/blog/using-oidc-github-actions-azure-swa/

**Reply by vignesh-gupta** [reply]
I see same as @feross and I've set them up for PR check. Any updates on this?

**Reply by kurt-rhee** [reply]
I seem to be having this issue as well, one of my secrets just doesn't show up

**Comment by fabian-michael-asi** [comment]
We encountered this today too. The secrets were suddenly missing in on of our PR workflows. Even after manually restarting. Eventually after committing another change it worked...
